lib/cisco_node_utils/interface.rb

```diff
@@ -1026,11 +1026,12 @@ def switchport_vtp_mode_capable?
     end
 
     def switchport_vtp
-      return nil unless switchport_vtp_mode_capable?
+      return default_switchport_vtp unless switchport_vtp_mode_capable?
```

Comment:
My argument was that `switchport_vtp` is not applicable to a non-switchport interface. We should thus report it as `nil`, just like we do for e.g. svi_autostate on non-Vlan interfaces.

Note that there's a TODO under `switchport_mode` that expresses a similar concern.

Another possibility to consider would be whether we really should be looking at splitting some of these interface types/modes into separate classes, similar to how interface_channel_group was extracted from interface to support only the subset of interface properties that are applicable to an interface in a channel-group.